### PR TITLE
Added Codec2 as a plug-in

### PR DESCRIPTION
**Pull request by baloian**
Thanks to @janakj it was possible to add [Codec2](http://www.rowetel.com/?page_id=452) library as a SEMS plug-in.

Current PR contains all modifications, which were done in order to add `Codec2` as a plug-in.

1. Created `codec2` directory inside of `${SEMS_ROOT}/core/plug-in/` directory.

2. Created `CMakeLists.txt`,  `codec2.c`,  `codec2.h` and `README.md` files inside of  `${SEMS_ROOT}/core/plug-in/codec2` directory.
- `codec2.h` file contains definition and `codec2.c` an implementation of Codec2 plug-in. 
- `README.md` file contains a basic info about Codec2 Later this may need to modify to provide more info.

Added Codec2 (3200, 2400, 1600 and 1400 bit rates) support in `${SEMS_ROOT}core/amci/codecs.h` file.
Modified `${SEMS_ROOT}/core/plug-in/codec2/CMakeLists.txt` in order to support codec2 plug-in building.

Note that we have a problem to install Codec2 (`libcodec2-dev`) on travis system

### CODE REVIEW

**Comment by baloian** [comment]
Hi @stefansayer would like you to know that current PR is ready to review (@janakj also is reviewing), but we have a problem to install Codec2 (`libcodec2-dev`) on `travis` system.

File: core/plug-in/codec2/codec2.c [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+ * codec2_bits_per_frame function in ${CODEC2_ROOT}/src/codec2.c file.
+ *
+ * It should be calculated like this (assuming that codec2 was already created):
+ *
+ * const int bits_per_frame = codec2_bits_per_frame(CODEC2);
+ * const int AMCI_FMT_ENCODED_FRAME_SIZE = (bits_per_frame + 7) / 8;
+ */
+static amci_codec_fmt_info_t codec2_fmt_description_20ms_3200[] = { {AMCI_FMT_FRAME_LENGTH, 20},
+                                                                    {AMCI_FMT_FRAME_SIZE, 160},
+                                                                    {AMCI_FMT_ENCODED_FRAME_SIZE, 8}, {0,0}};
```
Comment (janakj) [inline core/plug-in/codec2/codec2.c]: I would still like to find a way to obtain the number of bytes per payload from libcodec2. Although there is nothing wrong with hardcoding those numbers here, I am a little concerned about the numbers getting out of sync if libcodec2 is updated. In my option, there should be a single source of truth for parameters like these.

Comment (baloian) [reply core/plug-in/codec2/codec2.c]: I am working on it.

Comment (baloian) [reply core/plug-in/codec2/codec2.c]: Fixed, see https://github.com/columbia-irt/sems/blob/irt_lab_add_codec2/core/plug-in/codec2/codec2.c#L114-L123

**Comment by juha-h** [comment]
Is there still some work on this or is it ready for pull?

**Comment by janakj** [comment]
It's ready to pull.

**Comment by janakj** [comment]
On Wed, Jun 6, 2018 at 2:24 PM, juha-h <notifications@github.com> wrote:

> Jan Janak writes:
>
> > It's ready to pull.
>
> OK, I can do it if needed.
>

Please feel free to pull. Thanks a lot!

-Jan

**Pull request merged**